pkg/tfbridge/provider_test.go

```diff
@@ -3259,3 +3264,223 @@ func TestSchemaFuncsNotCalledDuringRuntime(t *testing.T) {
+			{
+			  "method": "/pulumirpc.ResourceProvider/Configure",
+			  "request": {
+				"args": {},
+				"variables": {}
+			  },
+			  "response": {
+				"supportsPreview": true
+			  }
+			},
+			{
+			  "method": "/pulumirpc.ResourceProvider/Check",
+			  "request": {
+				"urn": "urn:pulumi:dev::teststack::Res::exres",
+				"olds": {},
+				"news": {
+				},
+				"randomSeed": "iYRxB6/8Mm7pwKIs+yK6IyMDmW9JSSTM6klzRUgZhRk="
+			  },
+			  "response": {
+				"inputs": {
+				  "__defaults": [],
+				  "maxItemsOneProp": null
+				}
+			  }
+			}
+		  ]
+		  `)
+	})
+}
+
+func TestMinMaxItemsOneOptional(t *testing.T) {
+	p := testprovider.SchemaMinMaxItemsOneOptionalProvider()
+	shimProv := shimv2.NewProvider(p)
+	provider := &Provider{
+		tf:     shimProv,
+		config: shimv2.NewSchemaMap(p.Schema),
+		info: ProviderInfo{
+			P: shimProv,
+		},
+		resources: map[tokens.Type]Resource{
+			"Res": {
+				TF:     shimv2.NewResource(p.ResourcesMap["res"]),
+				TFName: "res",
+				Schema: &ResourceInfo{},
+			},
+		},
+	}
+
+	t.Run("No error when not specified", func(t *testing.T) {
+		testutils.ReplaySequence(t, provider, `[
+			{
+			  "method": "/pulumirpc.ResourceProvider/Configure",
+			  "request": {
+				"args": {},
+				"variables": {}
+			  },
+			  "response": {
+				"supportsPreview": true
+			  }
+			},
+			{
+			  "method": "/pulumirpc.ResourceProvider/Check",
+			  "request": {
+				"urn": "urn:pulumi:dev::teststack::Res::exres",
+				"olds": {},
+				"news": {
+				},
+				"randomSeed": "iYRxB6/8Mm7pwKIs+yK6IyMDmW9JSSTM6klzRUgZhRk="
+			  },
+			  "response": {
+				"inputs": {
+				  "__defaults": [],
+				  "maxItemsOneProp": null
+				}
+			  }
+			}
+		  ]
+		  `)
+	})
+
+	t.Run("No error when specified", func(t *testing.T) {
+		testutils.ReplaySequence(t, provider, `[
+			{
+			  "method": "/pulumirpc.ResourceProvider/Configure",
+			  "request": {
+				"args": {},
+				"variables": {}
+			  },
+			  "response": {
+				"supportsPreview": true
+			  }
+			},
+			{
+			  "method": "/pulumirpc.ResourceProvider/Check",
+			  "request": {
+				"urn": "urn:pulumi:dev::teststack::Res::exres",
+				"olds": {},
+				"news": {
+					"max_items_one_prop": ["prop"]
+				},
+				"randomSeed": "iYRxB6/8Mm7pwKIs+yK6IyMDmW9JSSTM6klzRUgZhRk="
+			  },
+			  "response": {
+				"inputs": {
+				  "__defaults": [],
+				  "maxItemsOneProp": "prop"
+				}
+			  }
+			}
+		  ]
+		  `)
+	})
+}
+
+func TestComputedMaxItemsOneNotSpecified(t *testing.T) {
+	p := testprovider.SchemaComputedMaxItemsOneNotSpecifiedProvider()
+	shimProv := shimv2.NewProvider(p)
+	provider := &Provider{
+		tf:     shimProv,
+		config: shimv2.NewSchemaMap(p.Schema),
+		info: ProviderInfo{
+			P: shimProv,
+		},
+		resources: map[tokens.Type]Resource{
+			"Res": {
+				TF:     shimv2.NewResource(p.ResourcesMap["res"]),
+				TFName: "res",
+				Schema: &ResourceInfo{},
+			},
+		},
+	}
+
+	t.Run("No error when plural specified", func(t *testing.T) {
```

Comment:
What does it mean "plural specified"?

Reply:
mb, this was copy pasted from a test I ended up deleting.